src/main/java/utilities/parser/PrescriptionValidator.java

```diff
@@ -173,4 +176,33 @@ public boolean isValidDate(Ui ui, String dateString) {
         }
         return false;
     }
+
+    /**
+     * Retrieves the total stock quantity for medicine with same name that has not expired.
+     *
+     * @param medicines     Arraylist of medicines.
+     * @param name          Medicine name.
+     * @param prescribeDate Date of Prescription.
+     * @return Total stock quantity for the medicine that has not expired.
+     */
+    public static int getNotExpiredStockQuantity(ArrayList<Medicine> medicines, String name, Date prescribeDate) {
+        int existingQuantity = 0;
+        for (Medicine medicine : medicines) {
+            if (!(medicine instanceof Stock)) {
+                continue;
+            }
+            boolean isSameMedicineName = medicine.getMedicineName().equalsIgnoreCase(name);
+            boolean isDeleted = ((Stock) medicine).isDeleted();
+            Date existingExpiry = ((Stock) medicine).getExpiry();
+            String expiryString = DateParser.dateToString(existingExpiry);
+            String prescribeDateString = DateParser.dateToString(prescribeDate);
+            boolean isNotExpired = (existingExpiry.after(prescribeDate) || prescribeDateString.equals(expiryString));
+
+            if (isSameMedicineName && !isDeleted && isNotExpired) {
+                existingQuantity += medicine.getQuantity();
+            }
+        }
+        return existingQuantity;
+    }
```

Comment:
Perhaps can put this method in `PrescriptionManager` class instead since it is not exactly a validator.